### ISSUE TITLE
Fulltext search

### ISSUE THREAD

**Issue by kaplun** [inspirehep/inspire-next#600]
Check this:
https://inspirehep.net/search?ln=en&ln=en&p=fulltext%3Agatis++0909.4255&of=hb&action_search=Search&sf=earliestdate&so=d&rm=&rg=25&sc=0

![sso1](https://cloud.githubusercontent.com/assets/119349/11527357/c926de12-98e0-11e5-8200-037159d389ef.png)

There are 2 problems:
* [ ] Stemming is enabled thus one can not search for the original "Gatis"
* [ ] The snippet seems mangled.

cc: @ksachs

**Comment by aw-bib** [comment]
Is there a way to enforce "as is no fancy stuff". Usually signified putting the string in quotes. (I think this would be helpful in general, not only for fulltext.)

**Comment by kaplun** [comment]
In this moment with elasticsearch everything is possible. As long as the user receives what she expects.

Note that with Elasticsearch we can be more Google like, hence be at the same time precise and fuzzy, but having the most precise results ranked on top.

Another wild idea is that we can expose to cataloguers the whole elasticsearch syntax and then you can search whatever you want:

https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html

**Comment by aw-bib** [comment]
Probably, I should be more direct: There should be a possibility to switch off all fancy stuff and search literally. Usually, this happens by quoting. This should be possible for mere mortals and cataloguers alike. It is _not_ a cataloguers need, but really common. And it should work the same in _all_ indices. (Just the question came up with the fulltext index here.)

My experience as a "searcher" with relevance ranking in scientific databases is "mixed", at best. Similarly, with "intelligent stuff". CAS e.g. can be a great resource, but sometimes you really just want to get rid off their fancy "natural language interface" and built in "intelligence". They don't have a switch for this. This can drive you nuts. Even on Google sometimes it is the only way to get to your result to switch off the intelligent parser by quoting your search term. At least I need this quite often.

**Comment by jmartinm** [comment]
Should we move this to the `inspire` repo instead of `inspire-next` to avoid confusion? As it is not currently a `bug` on Labs.

**Comment by kaplun** [comment]
Nope. Because on inspire repo I don't think it's possible to amend the fulltext search easily. While for Labs we are in time to do things right :)

**Comment by kaplun** [comment]
> There should be a possibility to switch off all fancy stuff and search literally. Usually, this happens by quoting. Usually, this happens by quoting. This should be possible for mere mortals and cataloguers alike. It is not a cataloguers need, but really common.

Yes: but the mere mortals are going to be confused if they start to combine several indexes... I better open a dedicated issue to discuss this.

**Comment by kaplun** [comment]
See #609
